configs/configs.go

```diff
@@ -2,25 +2,55 @@ package configs
 
 import (
 	"os"
+	"path/filepath"
 
 	"github.com/LetsFocus/goLF/logger"
 	"github.com/joho/godotenv"
 )
 
-func NewConfig(log *logger.CustomLogger) Config {
-	env := os.Getenv("APP_ENV")
-	envPath := ""
-	if env != "" {
-		envPath = "./configs/." + env + ".env"
-	} else {
-		envPath = "./configs/.env"
+func findConfigsDir(dir string) (string, error) {
+	depth := 0
```

Comment:
if depth always starts from zero, there is no base condition for recursion. The depth must be passed between the recursive calls. When the depth exceeds the maxDepth it is the base condition.